velodyne_pointcloud/src/lib/rawdata.cc

```diff
@@ -435,19 +436,27 @@ namespace velodyne_rawdata
             intensity = (intensity < min_intensity) ? min_intensity : intensity;
             intensity = (intensity > max_intensity) ? max_intensity : intensity;
     
+            // append this point to the cloud
+            VPoint point;
+            point.ring = corrections.laser_ring;
             if (pointInRange(distance)) {
-    
-              // append this point to the cloud
-              VPoint point;
-              point.ring = corrections.laser_ring;
+
               point.x = x_coord;
               point.y = y_coord;
               point.z = z_coord;
               point.intensity = (uint8_t) intensity;
-
-              pc.points.push_back(point);
-              ++pc.width;
             }
+            else {
+              
+              // If not in the specified range, set point to NaN.
+              point.x = std::numeric_limits<float>::infinity();
+              point.y = std::numeric_limits<float>::infinity();
+              point.z = std::numeric_limits<float>::infinity();
+              point.intensity = 0u;
+            }
+
+            pc.points.push_back(point);
```

Comment:
We take different approaches to organize the point clouds.
- You organize the points in RawData::unpack() and you do not provide support for the VLP-16.
- I hardly change RawData::unpack(), but I take the 1D point cloud it produces and organize it in [Convert](https://github.com/fudger/velodyne/blob/ordered_point_cloud_vlp16/velodyne_pointcloud/src/conversions/convert.cc#L84). My code only supports the VLP-16.

I would be happy if we could devise a general solution together! Perhaps we can build upon [organize_point_cloud](https://github.com/fudger/velodyne/blob/organize_point_cloud/velodyne_pointcloud/src/lib/rawdata.cc), a branch I started to organize point clouds coming from any Velodyne sensor? The clue in this branch is that RawData::unpack() takes Velodyne messages instead of individual packets.

PS: The idea of processing whole messages instead of packets is not only suited to organize point cloud. It comes in handy when [accurately transforming points](https://github.com/fudger/velodyne/blob/per_point_transform/velodyne_pointcloud/src/lib/rawdata.cc), too.